app.go

```diff
@@ -438,3 +449,90 @@ func executeCancelSell(cmd command) bool {
 
 	return true
 }
+
+func executeSetBuyAmount(cmd command) bool {
+	userID := cmd.UserID
+	strAmount := cmd.Args[1]
+	stock := cmd.Args[0]
+	account := accountStore.GetAccount(userID)
+
+	if account == nil {
+		log.Infof("User %s does not have an account", userID)
+		return false
+	}
+
+	amount, err := currency.NewFromString(strAmount)
+	if err != nil {
+		log.Error("Failed to parse currency")
+		return false
+	}
+	err = account.RemoveFunds(amount)
+	if err != nil {
+		log.Error("User had insufficient funds to set buy amount of %s", amount)
+		return false
+	}
+	autoBuyRequestStore.AddAutorequest(stock, cmd.UserID, amount)
+	log.Infof("User %s set automated buy amount for %s dollars of stock %s", userID, amount, stock)
+	return true
+}
+
+func executeSetSellAmount(cmd command) bool {
+	userID := cmd.UserID
+	strAmount := cmd.Args[1]
+	stock := cmd.Args[0]
+	account := accountStore.GetAccount(userID)
+
+	if account == nil {
+		log.Infof("User %s does not have an account", userID)
+		return false
+	}
+
+	amount, err := currency.NewFromString(strAmount)
+	if err != nil {
+		log.Error("Failed to parse currency")
+		return false
+	}
+	autoSellRequestStore.AddAutorequest(stock, userID, amount)
+	log.Infof("User %s set automated sell amount for %s dollars of stock %s", userID, amount, stock)
+	return true
+}
+
+func executeCancelSetBuy(cmd command) bool {
+	userID := cmd.UserID
+	stock := cmd.Args[0]
+	account := accountStore.GetAccount(userID)
+
+	if account == nil {
+		log.Infof("User %s does not have an account", userID)
+		return false
+	}
+
+	refund, err := autoBuyRequestStore.CancelAutorequest(stock, userID)
+	if err != nil {
+		log.Infof("Automated buy for stock %s was not found for user %s", stock, userID)
+	} else {
+		log.Infof("User %s cancelled automated buy for %s", userID, stock)
+		refundCurrency, _ := currency.NewFromFloat(refund) //should never error, so squelched the error
+		account.AddFunds(refundCurrency)
+	}
+	return true
+}
+
+func executeCancelSetSell(cmd command) bool {
+	userID := cmd.UserID
+	stock := cmd.Args[0]
+	account := accountStore.GetAccount(userID)
+
+	if account == nil {
+		log.Infof("User %s does not have an account", userID)
+		return false
+	}
+	_, err := autoSellRequestStore.CancelAutorequest(stock, userID)
+	if err != nil {
+		log.Infof("Automated sell for stock %s was not found for user %s", stock, userID)
```

Comment:
Do you want to return `true` from this point? `CancelAutoRequest` will error if an out-of-order set of operations is sent, which doesn't seem like it should have the same return status as "Everything was okay :+1:"